spec/core_ext/class/hierarchy_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'socket' # for a more interesting hierarchy for specs
+
+describe Class do
+  it "#hierarchy" do
+    expect(IO.hierarchy).to eq(
+      BasicSocket => {
```

Comment:
wasn't sure if `IO` itself should be in the hierarchy or not.  I put `self` in `lineage`.  Thoughts?